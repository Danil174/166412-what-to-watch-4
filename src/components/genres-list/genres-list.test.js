import React from "react";
import renderer from "react-test-renderer";
import GenresList from "./genres-list.jsx";

const genres = [`Comedies`, `Crime`, `Documentary`, `Dramas`, `Horror`, `Kids & Family`, `Romance`, `Sci-Fi`, `Thrillers`];
const activeGenre = `Comedies`;

describe(`GenresListTest`, () => {
  it(`render GenresList`, () => {
    const tree = renderer
      .create(
          <GenresList
            genres={genres}
            activeGenre={activeGenre}
            onGenreItemClick={() => {}}
          />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
