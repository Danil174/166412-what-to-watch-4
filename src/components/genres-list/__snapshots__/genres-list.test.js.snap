// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenresListTest render GenresList 1`] = `
<ul
  className="catalog__genres-list"
  onClick={[Function]}
>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      data-genre="Comedies"
      href="#"
    >
      Comedies
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      data-genre="Crime"
      href="#"
    >
      Crime
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      data-genre="Documentary"
      href="#"
    >
      Documentary
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      data-genre="Dramas"
      href="#"
    >
      Dramas
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      data-genre="Horror"
      href="#"
    >
      Horror
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      data-genre="Kids & Family"
      href="#"
    >
      Kids & Family
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      data-genre="Romance"
      href="#"
    >
      Romance
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      data-genre="Sci-Fi"
      href="#"
    >
      Sci-Fi
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      data-genre="Thrillers"
      href="#"
    >
      Thrillers
    </a>
  </li>
</ul>
`;
